File: 01-Array/283.moveZeroes.py
nums = [0,1,0,3,4]

def moveZeroes(nums: list[int]) -> list[int]:
    for i in range(len(nums)):
        if nums[i] == 0:
            nums.remove(0) # remove the first 0
            nums.append(0) # append 0 at the end
    return nums

print(moveZeroes(nums))

File: 01-Array/283.moveZeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        // bubble method
        int n = nums.size(), left = 0; 
        for(int right = 0; right < n; right++){
            if (nums[right] != 0){
                swap(nums[left], nums[right]); left++;
            }
        }
    }
};